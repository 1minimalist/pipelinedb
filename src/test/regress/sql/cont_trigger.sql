CREATE CONTINUOUS VIEW cont_trigger AS SELECT count(*) FROM stream;

-- Invalid triggers
CREATE TRIGGER cont_tg BEFORE INSERT ON cont_trigger FOR EACH ROW EXECUTE PROCEDURE pipeline_stream_insert();
CREATE TRIGGER cont_tg AFTER DELETE ON cont_trigger FOR EACH ROW EXECUTE PROCEDURE pipeline_stream_insert();
CREATE TRIGGER cont_tg AFTER INSERT ON cont_trigger FOR EACH STATEMENT EXECUTE PROCEDURE pipeline_stream_insert();
CREATE TRIGGER cont_tg AFTER TRUNCATE ON cont_trigger FOR EACH ROW EXECUTE PROCEDURE pipeline_stream_insert();
