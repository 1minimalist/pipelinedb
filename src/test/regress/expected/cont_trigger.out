CREATE CONTINUOUS VIEW cont_trigger AS SELECT count(*) FROM stream;
-- Invalid triggers
CREATE TRIGGER cont_tg BEFORE INSERT ON cont_trigger FOR EACH ROW EXECUTE PROCEDURE pipeline_stream_insert();
ERROR:  "cont_trigger" is a continuous view
DETAIL:  Continuous views can only have AFTER triggers.
CREATE TRIGGER cont_tg AFTER DELETE ON cont_trigger FOR EACH ROW EXECUTE PROCEDURE pipeline_stream_insert();
ERROR:  "cont_trigger" is not a sliding window continuous view
DETAIL:  Only sliding window continuous views can have DELETE triggers.
CREATE TRIGGER cont_tg AFTER INSERT ON cont_trigger FOR EACH STATEMENT EXECUTE PROCEDURE pipeline_stream_insert();
ERROR:  "cont_trigger" is a continuous view
DETAIL:  Continuous views can only have row-level AFTER triggers.
CREATE TRIGGER cont_tg AFTER TRUNCATE ON cont_trigger FOR EACH ROW EXECUTE PROCEDURE pipeline_stream_insert();
ERROR:  "cont_trigger" is a continuous view
DETAIL:  Continuous views cannot have TRUNCATE triggers.
